File: backend/src/main/java/com/back/fairytale/domain/fairytale/entity/FairytaleKeyword.java
package com.back.fairytale.domain.fairytale.entity;

import com.back.fairytale.domain.keyword.entity.Keyword;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "fairytale_keyword")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class FairytaleKeyword {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "fairytale_id", nullable = false)
    private Fairytale fairytale;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "keyword_id", nullable = false)
    private Keyword keyword;
}

File: backend/src/main/java/com/back/fairytale/domain/fairytale/entity/FairytaleKeyword.kt
package com.back.fairytale.domain.fairytale.entity

import com.back.fairytale.domain.keyword.entity.Keyword
import jakarta.persistence.*

@Entity
@Table(name = "fairytale_keyword")
class FairytaleKeyword(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "fairytale_id", nullable = false)
    val fairytale: Fairytale,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "keyword_id", nullable = false)
    val keyword: Keyword
)
